### PR TITLE
Fix, improve FYSETC_MINI_12864 support

### PR DESCRIPTION
**Pull request by jmdearras**
…check for light at boot.

### Requirements

Add support for FYSETC_MINI_12864 to BIGTREE_SKR_V1.3

### Benefits

Adds FYSETC_MINI_12864 display to SKRV1.3

### CODE REVIEW

**Comment by Bob-the-Kuhn** [comment]
> to have backlight enabled on startup

Please elaborate.  In my experience just connecting GND and +5V on EXP1 is enough to turn on the backlight.  Maybe the SKR is different than the card I have.

If the backlight startup is required then consider putting it in the pins file like you did the RGB LEDs.

**Comment by Bob-the-Kuhn** [comment]
I just looked at the 1.3  schematic.  No shared signals with EXP2 and the on-board SD card.  All EXP1 & EXP2 signal pins connected.  EXP1 & EXP2 have MKS sytle pin numbering (same as on the Fysetc LCD.

No special cabling. Nice.

**Comment by Bob-the-Kuhn** [comment]
Have you tested this with today's 2.0.x?  There was a commit merged last night that may cause problems.

I'm thinking line 115 in **ultralcd_DOGM.h** needs to change.
**FROM**
```cpp
     #if EITHER(FYSETC_MINI_12864, TARGET_LPC1768)
```
**TO**
```cpp
     #define IS_RE_ARM MB(RAMPS_14_RE_ARM_EFB) || MB(RAMPS_14_RE_ARM_EEB) || MB(RAMPS_14_RE_ARM_EFF) || MB(RAMPS_14_RE_ARM_EEF) || MB(RAMPS_14_RE_ARM_SF)
     #if ENABLED(FYSETC_MINI_12864) && (IS_RE_ARM || MB(MKS_SBASE))
```

File: Marlin/src/pins/pins_BIGTREE_SKR_V1.3.h [hunk 1]
```diff
@@ -190,28 +190,51 @@
...
+        #define DOGLCD_A0     P1_22
+      #endif
+
+      #if ENABLED(ULTIPANEL)
+        #define LCD_PINS_D5   P1_21
+        #define LCD_PINS_D6   P1_22
+        #define LCD_PINS_D7   P1_23
+      #endif
+
+    #endif // FYSETC_MINI_12864
```
Comment (thinkyhead) [inline Marlin/src/pins/pins_BIGTREE_SKR_V1.3.h:237]: I rearranged this to put `FYSETC_MINI_12864` within the `ULTRA_LCD` block where it belongs. I then removed pins that were redundant definitions. So this is the result according to the best information provided so far.

As a result of your changes,
- pin `P1_18` is assigned to both `LCD_PINS_ENABLE` and `DOGLCD_CS`,
- pin `P1_19` is assigned to both `LCD_PINS_RS` and `DOGLCD_A0`, and
- pin `P1_20` is assigned to both `LCD_PINS_D4` and `LCD_RESET_PIN`. 

If this was not your intention then more cleanup is needed.

Another aspect of this neophyte pins file is that for the `MKS_MINI_12864` pin `P1_21` is assigned to both `DOGLCD_CS` and `LCD_PINS_D5`. And pin `P1_22` is assigned to both `DOGLCD_A0` and `LCD_PINS_D6`. Hopefully this is not a problem either.

**Comment by thinkyhead** [comment]
@Bob-the-Kuhn — Your proposed change seems sensible, though it would be preferable for pins files to provide a flag instead of requiring `ultralcd_DOGM.h` to maintain expertise about boards.

**Comment by thinkyhead** [comment]
I think this is safe, and it correlates to the listed boards:
```cpp
  #if ENABLED(FYSETC_MINI_12864) && DOGLCD_SCK > 0
```

File: Marlin/src/pins/pins_BIGTREE_SKR_V1.3.h (hunk 1 above)
Comment (thinkyhead) [reply Marlin/src/pins/pins_BIGTREE_SKR_V1.3.h]: Tweaked that last bit by changing the condition for `LCD_PINS_*` to a character LCD. (The `ULTIPANEL` could be either character or graphical.)

**Comment by Bob-the-Kuhn** [comment]
Assigning the logical pins to multiple names is OK in this case.  That's because the functions that use the different names are mutually exclusive.

I'll pull this PR down and test on my boards late tonight or tomorrow morning.

**Comment by jmdearras** [comment]
This patch was written by a FB friend, Brian Ipsen, and I was going to test it myself before submitting it for him.   However, my display,, ordered mid-Feb,, seems to be lost in the mail.  I have sent Brian this URL to review and comment

**Comment by bipsendk** [comment]
Tried to compile with the updated/committed pins_BIGTREE_SKR_V1.3.h and SanityCheck.h - result is OK.
If LED_USER_PRESET_STARTUP not has been defined, the Fysetc will have no LED backlight a all (at least on my 2 SKR v1.3 boards and 2 Fysetc Mini 12864 v2.0 Model B displays).
A v2.1 version of the display has been introduced by the vendor recently, where the LED's are controlled via Neopixel API. Not sure if that version would require an additional define in order to sanity check whether Neopixel has been enabled. - Unfortunately I am unable to test, as I don't have any v2.1 displays..

**Comment by jmdearras** [comment]
Brian Ipsen: "Code seems OK - have just tested with the committed/changed files..."

**Comment by Bob-the-Kuhn** [comment]
Doesn't work on RAMPS (mega2560), RE-ARM or Due.

I'll investigate further.

**Comment by Bob-the-Kuhn** [comment]
Good news & bad news.

I screwed up earlier.  The code on this PR works on all three once line 1792 in **SanityCheck.h** is changed.
**FROM**

**TO**

**Comment by Bob-the-Kuhn** [comment]
Good news & bad news.

I screwed up earlier. The code on this PR works on all three once line 1792 in **SanityCheck.h** is changed.
**FROM**
`   #if ENABLED(FYSETC_MINI_12864) && DISABLED(LED_USER_PRESET_STARTUP) `
**TO**
`   #if ENABLED(FYSETC_MINI_12864) && DISABLED(LED_USER_PRESET_STARTUP) && MB(BIGTREE_SKR_V1_3)`

The bad news is enabling the SD card crashes RAMPS and DUE when the SD card is inserted.

I suggest proceeding with merging this PR (after the **SanityCheck.h** fix is done) and pursuing the SD card problem under a new PR.

**Comment by bipsendk** [comment]
@Bob-the-Kuhn There is an issue with that change - because if LED_USER_PRESET_STARTUP is omitted on a Fysetc F6 v1.3 board, the display will have no backlight (just as with the SKR board). Having a display with "white" text on a black background will be very hard to read without backlight,

But I can confirm, that the F6 also crashes when a SD card is inserted on the Fysetc Mini 12864 ...

**Comment by bipsendk** [comment]
FYI: I have a build from 20190314 where I added support in pins.h file for the Fysetc F6 - and in that one there are no issues with the F6 board and the display when I insert a SD card (except that the display blanks for a short moment and then comes up again with "SD Card inserted)

**Comment by Bob-the-Kuhn** [comment]
The proposed change limits the LED_USER_PRESET_STARTUP error check to just the F6 V1.3.

**Comment by thinkyhead** [comment]
> The proposed change limits the LED_USER_PRESET_STARTUP error check to just the F6 V1.3.

If a sanity check against a hardware requirement is going to only pertain to a specific board, I propose that the sanity check be located in the board's pins file.

**Comment by thinkyhead** [comment]
Since the original branch for this PR has been deleted, I can either close it and create a new one, or merge this and create a followup with the remaining changes. So, I will merge this and push the changes shortly.

**Comment by thinkyhead** [comment]
> The proposed change limits the LED_USER_PRESET_STARTUP error check to just the F6 V1.3.

Looks like it adds the check for `BIGTREE_SKR_V1_3`, not the `F6` board. Which is correct?

**Comment by bipsendk** [comment]
@thinkyhead LED_USER_PRESET_STARTUP is required on both SKR v1.3 and F6 - that is at least what I can see on the two boards, where I have tested the display.

**Comment by Bob-the-Kuhn** [comment]
**EVERYONE** - PR #13767 is a follow up  Please review and test.

I'm surprised at the changes to the FYSETC pins files.  I'm concerned I have outdated copies.  Please look them over carefully.

**Comment by Bob-the-Kuhn** [comment]
This should catch just the two boards:
```
#if ENABLED(FYSETC_MINI_12864) && DISABLED(LED_USER_PRESET_STARTUP) && EITHER(MB(BIGTREE_SKR_V1_3), MB(FYSETC_F6_13))
  #error "FYSETC_MINI_12864 requires LED_USER_PRESET_STARTUP to enable the backlight on startup."
#endif
```

**Comment by bipsendk** [comment]
Where should i download from - if I should try to compile for SKR v1.3 and Fysetc F6 (both with Fysetc Mini 12864 v2.0) ?

**Pull request merged**